recipes/vvenc/all/conanfile.py

```diff
@@ -0,0 +1,120 @@
+import os
+import sys
+
+from conan import ConanFile, conan_version
+# FIXME: linter complains, but function is there
+# https://docs.conan.io/2.0/reference/tools/build.html?highlight=check_min_cppstd#conan-tools-build-check-max-cppstd
+# from conan.tools.build import stdcpp_library, check_min_cppstd, check_max_cppstd
+from conan.tools.build import stdcpp_library, check_min_cppstd
+from conan.tools.cmake import CMakeToolchain, CMake, cmake_layout, CMakeDeps
+from conan.tools.files import get, copy, rmdir, rm
+
+required_conan_version = ">=1.60.1"
+
+
+class vvencRecipe(ConanFile):
+    name = "vvenc"
+    license = "BSD-3-Clause"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "Fraunhofer Versatile Video Encoder (VVenC)"
+    topics = ("video", "encoder", "codec", "vvc", "h266")
+    homepage = "https://www.hhi.fraunhofer.de/en/departments/vca/technologies-and-solutions/h266-vvc.html"
+    package_type = "library"
+
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+
+    def validate_build(self):
+        if conan_version.major == 2:
+            self._validate_build2()
+        elif conan_version.major == 1:
+            self._validate_build1()
+
+    def _validate_build1(self):
+        if self.settings.get_safe("compiler.cppstd"):
+            check_min_cppstd(self, 14)
+        # compiler.cppstd isn't set! but we still continue here in V1,
+        # imagining just like compiler.cppstd was set to 14
+        # while package_id doesn't reflect that at all, and cppstd_default also
+        # might be different from 14, we silently force -std=c++14 to be
+        # specified during the build. it may produce more incompatibilities,
+        # and break user's expectation
+        # (like, output binary depends really on C++14 symbols libstdc++.so)
+        # it's V1 design flaw which isn't going to be addressed here
+        # (and probably nowhere, because conan V1 is going to be discontinued in CCI)
+        # once V1 is retired, that code will be removed altogether
+        self.output.warning("compiler.cppstd is not set, but we assume C++14")
+
+    def _validate_build2(self):
+        # validates the minimum and maximum C++ standard supported
+        # currently, the project can only be built with C++14 standard
+        # it cannot be built with older standard because
+        # it doesn't have all the C++ features needed
+        # and it cannot be built with newer C++ standard
+        # because they have existing C++ features removed
+        check_min_cppstd(self, 14)
+        # FIXME: linter complains, but function is there
+        # https://docs.conan.io/2.0/reference/tools/build.html?highlight=check_min_cppstd#conan-tools-build-check-max-cppstd
+        check_max_cppstd = getattr(sys.modules['conan.tools.build'], 'check_max_cppstd')
+        check_max_cppstd(self, 14)
```

Comment:
not sure why the linter complains about this but I'd rather have the wrong warning than importing the function this way.

Also believe there is no need to check conditionally to the conan version, and that the validate method could be unified:

```
def validate(self):
    if self.settings.get_safe("compiler.cppstd"):
        check_min_cppstd(self, 14)
        check_max_cppstd(self, 14)
    else:
        self.output.warning("compiler.cppstd is not set, but we assume C++14")
```